### DISCUSSION TITLE
unauthorized: unauthenticated: User cannot be authenticated with the token provided.

### DISCUSSION THREAD

**Discussion by sebastian-luna-valero** [community/community#50035] [Packages] [answered]
### Select Topic Area

Question

### Body

Hello,

Following steps in https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry

I created a new Personal Access Token with scopes:
* `read:packages`
* `read:packages`
* `delete:packages`

Then:
```
echo $CR_PAT | docker login ghcr.io -u USERNAME --password-stdin
```

All good so far, but then:

```
sudo docker run hello-world
sudo docker tag hello-world ghcr.io/c-scale-community/hello-world
sudo docker push ghcr.io/c-scale-community/hello-world
```
I get
```
Using default tag: latest
The push refers to repository [ghcr.io/c-scale-community/hello-world]
e07ee1baac5f: Preparing 
unauthorized: unauthenticated: User cannot be authenticated with the token provided.
```

I must be missing something basic, but I don't see what.

Could you please help?

Many thanks,
Sebastian

**Comment by sebastian-luna-valero** [comment, marked as answer]
I found the issue:
```bash
echo $CR_PAT | docker login ghcr.io -u USERNAME --password-stdin
```
in https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry

Should be:
```bash
echo $CR_PAT | sudo docker login ghcr.io -u USERNAME --password-stdin
```
Note `sudo` in the latter command.

With the first command, `${HOME}/.docker/config.json` is populated but not picked up by `sudo docker push ghcr.io/c-scale-community/hello-world`

With the second command `/root/.docker/config.json` is populated and correctly picked up by `sudo docker push ghcr.io/c-scale-community/hello-world`

**Reply by daye1997** [reply]
Thank you so much, that fixed my problem. !

**Reply by waniilo** [reply]
💯
